src/fu-udev-backend.c

```diff
@@ -335,6 +392,63 @@ fu_udev_backend_uevent_cb(GUdevClient *gudev_client,
 	}
 }
 
+static void
+fu_udev_backend_coldplug_subsystem_by_path_internal(FuUdevBackend *self, const gchar *fn)
+{
+	const gchar *basename;
+	g_autoptr(GDir) dir = NULL;
+	g_autoptr(GError) error_dir = NULL;
+
+	dir = g_dir_open(fn, 0, &error_dir);
+	if (dir == NULL) {
+		if (g_error_matches(error_dir, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+			return;
+		g_debug("ignoring: %s", error_dir->message);
+		return;
```

Comment:
you can drop an extra return statement if you swap it like this:

```suggestion
		if (!g_error_matches(error_dir, G_FILE_ERROR, G_FILE_ERROR_NOENT))
			g_debug("ignoring: %s", error_dir->message);
		return;
```

Reply:
Agreed, i've used your suggestion, thanks.